scripts/fill_dcp1_file_metadata/fill_dcp1_metadata.py

```diff
@@ -0,0 +1,249 @@
+    """
+    query = {
+        "projectId": {
+            "is":
+                [
+                    project_uuid
+                ]
+        },
+        "fileSource": {
+            "is":
+                [
+                    None,
+                    "Contributor",
+                    "ArrayExpress",
+                    "GEO",
+                    "SCEA",
+                    "SCP",
+                    "LungMAP",
+                    "Zenodo",
+                    "Publication"
+                ]
+        }
+    }
+    return json.dumps(query)
+
+
+def yield_pages(first_page, session):
+    """
+    There has to be a more elegant way to deal with pagination
+    :param first_page:
+    :param session:
+    :return:
+    """
+    next_page = session.get(first_page['pagination']['next']).json()
+    yield next_page
+    while next_page['pagination'].get('next'):
+        next_page = session.get(next_page['pagination']['next']).json()
+        yield next_page
+
+
+def get_files_metadata_from_azul(azul_base: str, query: str) -> list:
+    """
+    Retrieve file metadata from azul
+    :param azul_base:       Base URL of Azul's API. Defaults to prod. From config.
+    :param query:           Query in str format to pass as a filter
+    :return files:          List of file metadata
+    """
+    # Build URL/endpoint and request metadata
+    azul_manifest_request_endpoint = "index/files"
+    session = rq.Session()
+
+    # Get info from pagination with minimal load (1 entity)
+    r = session.get(f"{azul_base}{azul_manifest_request_endpoint}?filters={query}&size=100")
+    r.raise_for_status()
+    r = r.json()
+
+    hits = r['hits']
+    for page in yield_pages(r, session):
+        hits.extend(page['hits'])
+
+    # Document ID (file metadata UUID) is outside of ["hits"]["files"], so some data massage
+    def change_datafile_uuid(azul_hit: dict):
+        azul_hit['files'][0]['uuid'] = azul_hit['entryId']
+        return azul_hit
+
+    files = list(map(change_datafile_uuid, hits))
+
+    # A bit more data massage to return only the file metadata (Easier parsing for later)
+    files = [hit['files'][0] for hit in files]
+
+    return files
+
+
+def parse_azul_metadata(azul_file_list: list) -> dict:
+    """
+    Parse the azul metadata about the files and return a map {uuid: metadata}, excluding DCP1/2 analysis files
+    :param azul_file_list:          File list from azul, output from get_files_metadata_from_azul()
+    :return file_metadata_by_uuid:  Dictionary in the format {file_uuid:file_metadata} for all files in azul_file_list
+    """
+    uuid_metadata_map = {file['uuid']: file for file in azul_file_list}
+
+    filesource_invalid_values = ["HCA Release", "DCP/2 Analysis", "DCP/1 Matrix Service"]
+    uuid_metadata_map_filtered = {uuid: metadata for uuid, metadata in uuid_metadata_map.items()
+                                  if not any([source == metadata['fileSource'] for source in filesource_invalid_values])}
```

Comment:
There is a simpler way (also more efficient) of doing it.
Probably in small scales the difference in runtime is hardly noticeable, but still, it is a good practice to use efficient algorithms.
```suggestion
    uuid_metadata_map_filtered = {uuid: metadata for uuid, metadata in uuid_metadata_map.items()
                                  if metadata['fileSource'] not in filesource_invalid_values}
```